### PR TITLE
add tests for ChefFS path formatting

### PR DESCRIPTION
**Pull request by thommay**
@chef/client-core

### CODE REVIEW

File: spec/unit/chef_fs/config_spec.rb [hunk 1]
```diff
@@ -103,4 +103,133 @@ def platform_path(*args)
...
+
+      it "returns the entry's path if the cwd isn't in the config" do
+        cfg = Chef::ChefFS::Config.new(config, "/my_repo/cookbooks")
+        expect(cfg).to receive(:base_path).and_return(nil)
+        expect(cfg.format_path(entry)).to eq(path)
+      end
+
+      it "returns . if the cwd is the same as the entry's path" do
+        cfg = Chef::ChefFS::Config.new(config, "/base_path/roles/foo.json")
+        expect(cfg).to receive(:base_path).and_return("/roles/foo.json").exactly(4).times
```
Comment (randomcamel) [inline spec/unit/chef_fs/config_spec.rb]: I usually see `exactly` as a test smell: say it breaks and someone looks at the test, is there an intuitive way to connect the number of times `base_path` gets called with what the code is doing? (Especially if our hypothetical debugger looks at the next 2 tests and sees different numbers for `exactly`.)

Comment (thommay) [inline spec/unit/chef_fs/config_spec.rb]: fair, can just make these `at_least`.

**Comment by thommay** [comment]
@chef/client-core @randomcamel ready for more review thank you, please

**Comment by lamont-granquist** [comment]
:+1:

**Pull request merged**